huntsman/utils/config.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Jan  2 12:04:11 2020
+
+@author: danjampro
+
+Code to provide a config server using pyro.
+"""
+import os, time
+import Pyro4
+from huntsman.utils import load_config, get_own_ip
+
+#==============================================================================
+
+@Pyro4.expose
+class ConfigServer():
+    '''
+    Class representing the config server.
+    '''
+    def __init__(self, config_file=None, **kwargs):
+        '''
+        
+        '''
+        if config_file is None:
+            config_file = os.path.join(os.environ['HUNTSMAN_POCS'],
+                                       'conf_files', 'camera_info_local.yaml')
+            
+        #Read the config file
+        self.config = load_config(config_files=[config_file], **kwargs)
+        
+        
+    def get_config(self, key=None):
+        '''
+        Retrieve the config file.
+        '''
+        if key is None:
+            return self.config
+        return self.config[key]
+    
+#==============================================================================
+
+def locate_name_server(wait=None, logger=None):
+    '''
+    Locate and return the name server, waiting if necessary.
+    
+    Parameters
+    ----------
+    wait (float or None) [seconds]:
+        If not None, attempt to locate the NS at this frequency. 
+
+    Returns
+    -------
+    Pyro name server.        
+    '''
+    if wait is None:
+        return Pyro4.locateNS()
+    
+    while True:
+        try:
+            return Pyro4.locateNS()
+        except Pyro4.errors.PyroError:
+            msg = 'Unable to locate name server. Waiting...'
+            if logger is None:
+                print(msg)
+            else:
+                logger.debug(msg)
+            time.sleep(wait)
+        
```

Comment:
Do you want to catch a `KeyboardInterrupt` so someone can cmd+c cancel this cleanly? Maybe returning a None so the routine calling it knows it should shut down?

Reply:
Sounds good, but at the moment I don't think there would be much difference. I also think returning None rather than raising an error might cause more confusion down the line. Do you have a specific scenario where this would be useful in-mind?

Reply:
Agree returning None is arbitrary and could cause issues @danjampro . How cntrl+C triggers an `sys.exit(0)` call with a kind message saying shutting down name server client.

Reply:
Oh and make sure to send the shutdown message to the logger as well @danjampro